EU4toV2/Source/V2World/Army/Army.cpp

```diff
@@ -281,6 +283,10 @@ V2::AddRegimentToArmyResult V2::Army::addRegimentToArmy(const REGIMENTTYPE chose
 		// Assign a national name ("1st Bavarian Frigate")
 		regiment.setName(getRegimentName(chosenType, unitNameCount, localAdjective));
 	}
+	if (unitNames)
+	{
+		regiment.nameShip(unitNames);
+	}
```

Comment:
@Zemurin This is all the change. Everything else is just importing/outputting unit names and helper functions for setting the ship name.